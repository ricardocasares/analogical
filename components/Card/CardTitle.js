// external
import React from "react";

export default ({ theme, children }) => (
  <h4>
    {children}
    <style jsx>{`
      h4 {
        color: black;
        margin: 0;
        font-weight: 400;
        font-size: 1.4rem;
        padding-bottom: 15px;
        margin-bottom: 15px;
        border-bottom: 1px solid ${theme.card.border};
      }
    `}</style>
  </h4>
);
