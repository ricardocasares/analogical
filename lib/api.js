// external
import fetch from "isomorphic-unfetch";

// internal
import cfg from "../config";

const githubClient = (api, token) => endpoint =>
  fetch(`${api}${endpoint}`, {
    headers: {
      Authorization: `Bearer ${token}`
    }
  }).then(res => res.json());

const gh = githubClient(cfg.GITHUB_API, cfg.GITHUB_TOKEN);

const user = endpoint => gh(`/users/${cfg.GITHUB_OWNER}${endpoint}`);
const repo = endpoint =>
  gh(`/repos/${cfg.GITHUB_OWNER}/${cfg.GITHUB_REPO}${endpoint}`);

export const getIssue = id => repo(`/issues/${id}`);
export const getIssues = (label = "") => repo(`/issues?labels=${label}`);
export const getLabels = () => repo(`/labels`);
export const getComments = id => repo(`/issues/${id}/comments`);
export const getLatestRepos = () => user(`/repos?per_page=6&sort=updated`);
export const getLatestSongs = () => forRecentSongs(5);
