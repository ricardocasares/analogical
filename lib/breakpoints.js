export const MOBILE = "min-width: 425px";
export const TABLET = "min-width: 600px";
export const DESKTOP = "min-width: 1024px";
