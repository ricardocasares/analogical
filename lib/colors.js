export const BLACK = "#111111";
export const WHITE = "#ffffff";
export const GRAY = "#AAAAAA";
export const LIGHT_GRAY = "#DDDDDD";
export const BLUE = "#0074D9";
export const PURPLE = "#B10DC9";
export const ORANGE = "#FF851B";
