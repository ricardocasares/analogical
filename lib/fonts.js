export const SANS = [
  "-apple-system",
  "BlinkMacSystemFont",
  "avenir next",
  "avenir",
  "helvetica",
  "helvetica neue",
  "ubuntu",
  "roboto",
  "noto",
  "segoe ui",
  "arial",
  "sans-serif"
].join(", ");

export const MONO = [
  "'SFMono-Regular'",
  "Consolas",
  "'Liberation Mono'",
  "Menlo",
  "Courier",
  "monospace"
].join(", ");
