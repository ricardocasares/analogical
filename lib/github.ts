import fetch from "isomorphic-unfetch";
import memoize from "memoize-promise";

const EXPIRE = 600000;
const GITHUB_TOKEN = process.env.GITHUB_TOKEN || "";
const GITHUB_GQL_ENDPOINT = process.env.GITHUB_GQL_ENDPOINT || "";

export const client = memoize(
  (query: string, variables: Object) =>
    request(query, variables, GITHUB_TOKEN, GITHUB_GQL_ENDPOINT),
  EXPIRE
);

export const request = async (
  query = "",
  variables = {},
  token = GITHUB_TOKEN,
  endpoint = GITHUB_GQL_ENDPOINT
) =>
  await fetch(endpoint, {
    method: "POST",
    body: JSON.stringify({
      query,
      variables
    }),

    headers: {
      authorization: `Bearer ${token}`
    }
  })
    .then(checkResponse)
    .then(res => res.json())
    .then(({ data }) => data);

function checkResponse(res: Response): Response {
  if (!res.ok) {
    throw new Error(`Code: ${res.status}`);
  }

  return res;
}
