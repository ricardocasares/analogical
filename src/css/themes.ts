import { createTheme } from "@/css";
import { slateDark, indigoDark } from "@radix-ui/colors";

export const dark = createTheme({
  colors: {
    ...slateDark,
    ...indigoDark,
  },
});
