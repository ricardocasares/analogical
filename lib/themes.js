// external
import c from "open-color";

// internal
import { SANS, MONO } from "./fonts";

export const dark = {
  global: {
    time: c.gray[4],
    code: c.gray[6],
    text: c.gray[7],
    background: c.gray[0],
    anchors: c.gray[7],
    blockquote: c.gray[7],
    strong: c.gray[9]
  },
  logo: c.gray[9],
  pic: {
    background: c.gray[8],
    loading: [c.gray[8], c.gray[7]]
  },
  posts: {
    list: c.gray[2]
  },
  card: {
    title: c.gray[9],
    hover: c.gray[5],
    border: c.gray[3]
  },
  type: {
    title: c.gray[8],
    muted: c.gray[7],
    fonts: {
      sans: SANS,
      mono: MONO
    }
  },
  chart: {
    title: c.gray[9],
    subtitle: c.gray[6],
    colors: [
      c.pink[6],
      c.grape[6],
      c.violet[6],
      c.indigo[6],
      c.blue[6],
      c.cyan[6],
      c.teal[6],
      c.green[6],
      c.lime[6],
      c.yellow[6],
      c.orange[6],
      c.red[6]
    ]
  },
  loading: c.violet[6],
  codeblock: {
    color: c.gray[7],
    background: c.gray[2]
  },
  comment: {
    author: c.gray[4],
    border: c.gray[7]
  }
};

export const cyan = {
  loading: c.cyan[6],
  logo: c.cyan[6]
};

export const teal = {
  loading: c.teal[6],
  logo: c.teal[6]
};

export const orange = {
  loading: c.orange[6],
  logo: c.orange[6]
};

export const blue = {
  loading: c.blue[6],
  logo: c.blue[6]
};
