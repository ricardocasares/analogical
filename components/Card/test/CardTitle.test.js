// external
import renderer from "react-test-renderer";

// internal
import { dark } from "lib/themes";
import CardTitle from "../CardTitle";

describe("CardTitle", () => {
  it("CardTitle", () => {
    const title = renderer
      .create(<CardTitle theme={dark}>Hello</CardTitle>)
      .toJSON();

    expect(title).toMatchSnapshot();
  });
});
