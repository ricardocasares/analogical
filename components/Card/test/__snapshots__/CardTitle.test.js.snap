// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTitle CardTitle 1`] = `
<h4
  className="jsx-3680783396"
>
  Hello
</h4>
`;
